zilliqa/src/transaction.rs

```diff
@@ -317,6 +320,34 @@ impl SignedTransaction {
             }
         }
     }
+
+    pub fn validate_gas_limit(&self) -> Result<()> {
+        // Verify minimum gas requirement
+        match self {
+            SignedTransaction::Legacy { .. }
+            | SignedTransaction::Eip2930 { .. }
+            | SignedTransaction::Eip1559 { .. } => {
+                if self.gas_limit() < MINIMUM_ETH_GAS {
+                    bail!(
+                        "Provided evm transaction gas limit is too low: {}",
+                        self.gas_limit()
+                    );
+                }
+            }
+            SignedTransaction::Zilliqa { ref tx, .. } => {
+                // Minimum gas needed for scilla run - contract invocation
+
+                if tx.gas_limit < MINIMUM_ZIL_GAS {
```

Comment:
Didn't we agree to move these changes to a separate PR (fixing #1003) so that we can double check this minimum is correct and doesn't break existing mainnet transactions?